api/v1alpha1/jvmchaos_types.go

```diff
@@ -0,0 +1,294 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// JVMChaosSpec defines the desired state of JVMChaos
+type JVMChaosSpec struct {
+	// Mode defines the mode to run chaos action.
+	// Supported mode: one / all / fixed / fixed-percent / random-max-percent
+	Mode PodMode `json:"mode"`
+
+	// Value is required when the mode is set to `FixedPodMode` / `FixedPercentPodMod` / `RandomMaxPercentPodMod`.
+	// If `FixedPodMode`, provide an integer of pods to do chaos action.
+	// If `FixedPercentPodMod`, provide a number from 0-100 to specify the max % of pods the server can do chaos action.
+	// If `RandomMaxPercentPodMod`,  provide a number from 0-100 to specify the % of pods to do chaos action
+	// +optional
+	Value string `json:"value"`
+
+	// Selector is used to select pods that are used to inject chaos action.
+	Selector SelectorSpec `json:"selector"`
+
+	// Duration represents the duration of the chaos action
+	// +optional
+	Duration *string `json:"duration,omitempty"`
+
+	// Scheduler defines some schedule rules to control the running time of the chaos experiment about time.
+	// +optional
+	Scheduler *SchedulerSpec `json:"scheduler,omitempty"`
+
+	// Action defines the specific jvm chaos action.
+	// Supported action: delay, return, script, cfl, oom, ccf, tce, delay4svr, tce4svr
+	// +kubebuilder:validation:Enum=delay;return;script;cfl;oom;ccf;tce;delay4svr;tce4svr
+	Action JVMChaosAction `json:"action"`
+
+	// JVMParmeter represents the detail about jvm chaos action definition
+	// +optional
+	JVMParmeter `json:",inline"`
+}
+
+// JVMChaosAction represents the chaos action about jvm
+type JVMChaosAction string
+
+const (
+	// JVMDelayAction represents the JVM chaos action of invoke delay
+	JVMDelayAction JVMChaosAction = "delay"
+
+	// JVMReturnAction represents the JVM chaos action of return value
+	JVMReturnAction JVMChaosAction = "return"
+
+	// JVMReturnAction represents the JVM chaos action for complex failure scenarios.
+	// Write Java or Groovy scripts, such as tampering with parameters, modifying return values,
+	// throwing custom exceptions, and so on
+	JVMScriptAction JVMChaosAction = "script"
+
+	// JVMCpuFullloadAction represents the JVM chaos action of CPU is full
+	JVMCpuFullloadAction JVMChaosAction = "cfl"
+
+	// JVMOOMAction represents the JVM chaos action of OOM exception
+	JVMOOMAction JVMChaosAction = "oom"
+
+	// JVMCodeCacheFillingAction represents the JVM chaos action of code cache filling
+	JVMCodeCacheFillingAction JVMChaosAction = "ccf"
+
+	// JVMExceptionAction represents the JVM chaos action of throwing custom exceptions
+	JVMExceptionAction JVMChaosAction = "tce"
```

Comment:
what is the mean of `tce`?

Reply:
throwing custom exceptions😂

Any Suggestions?

Reply:
just use `exception`? I don't have any better ideas.

Reply:
Reference to [  chaosblade doc](https://chaosblade-io.gitbook.io/chaosblade-help-zh-cn/blade-create-jvm/blade-create-jvm-throwcustomexception), tce is an abbreviation.  I don't think it's a traditional abbreviation, but it's ok to use it. Because we are implementing JVM chaos based on it.

Reply:
ok